### PR TITLE
Correct Missing Type Annotations

### LINKED ISSUE #9: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_sht4x.py:49
- [ ] adafruit_sht4x.py:61
- [ ] adafruit_sht4x.py:129
- [ ] adafruit_sht4x.py:169
- [ ] adafruit_sht4x.py:229

### PR DESCRIPTION
**Pull request by tcfranks**

### CODE REVIEW

**Comment by tcfranks** [comment]
ready for review / comments

**Comment by tekktrik** [comment]
I can take a look at this tonight, but see other libraries that use the `CV` class implementation for help on type annotating this:  https://github.com/adafruit/Adafruit_CircuitPython_TLA202x/blob/main/adafruit_tla202x/__init__.py

File: adafruit_sht4x.py [hunk 1]
```diff
@@ -46,7 +52,7 @@ class CV:
     """struct helper"""
 
     @classmethod
-    def add_values(cls, value_tuples):
+    def add_values(cls, value_tuples) -> None:
```
Comment (tekktrik) [inline adafruit_sht4x.py]: See https://github.com/adafruit/Adafruit_CircuitPython_TLA202x/blob/main/adafruit_tla202x/__init__.py for help annotating `CV.add_values()`, as they work identically.

File: adafruit_sht4x.py [hunk 2]
```diff
@@ -58,7 +64,7 @@ def add_values(cls, value_tuples):
             cls.delay[value] = delay
 
     @classmethod
-    def is_valid(cls, value):
+    def is_valid(cls, value) -> bool:
```
Comment (tekktrik) [inline adafruit_sht4x.py]: See https://github.com/adafruit/Adafruit_CircuitPython_TLA202x/blob/main/adafruit_tla202x/__init__.py for help annotating `CV.is_valid()`, as they work identically.

**Review by tekktrik — CHANGES_REQUESTED** [review]
Everything looks good so far!  I provided a link for help with type annotating the remaining parts.

**Comment by tcfranks** [comment]
ready for re-review

File: adafruit_sht4x.py [hunk 3]
```diff
@@ -46,7 +52,9 @@ class CV:
     """struct helper"""
 
     @classmethod
-    def add_values(cls, value_tuples):
+    def add_values(
+        cls, value_tuples: Tuple[str, int, Union[float, str], Optional[float]]
```
Comment (tekktrik) [inline adafruit_sht4x.py]: In this instance, it looks like it's `Tuple[str, int, str, float]`

File: adafruit_sht4x.py [hunk 4]
```diff
@@ -58,7 +66,7 @@ def add_values(cls, value_tuples):
             cls.delay[value] = delay
 
     @classmethod
-    def is_valid(cls, value):
+    def is_valid(cls, value: str) -> bool:
```
Comment (tekktrik) [inline adafruit_sht4x.py]: Looks like the base material I gave you was wrong, `value` should be `int`

**Review by tekktrik — CHANGES_REQUESTED** [review]
Thanks for adding the remaining things!  It looks like the base material was a little off or could be made slightly more specific here so a few changes for that.

**Review by tekktrik — APPROVED** [review]
Thanks!

**Comment by tcfranks** [comment]
review for prev comments

**Pull request merged**